syntax = "proto3";

import "eigen.proto";

message UnivariateMixtureAtom {
  double mean = 1;
  double stdev = 2;
}

message UnivariateMixtureState {
    int32 num_components = 1;
    repeated double weights = 2;
    repeated UnivariateMixtureAtom atoms = 3;
}

message UnivariateState {

  message GroupWeights {
    repeated double weights = 1;
  }

  int32 num_components = 1;
  repeated UnivariateMixtureAtom atoms = 2;
  repeated GroupWeights groupWeights = 3;
  double rho = 4;
  EigenMatrix Sigma = 5;
}
